src/ProductBundle/Entity/BaseProduct.php

```diff
@@ -245,7 +246,9 @@ public function getSku()
      */
     public function setSlug($slug)
     {
-        $this->slug = self::slugify(trim($slug));
+        $slug = Slugify::create()->slugify(trim($slug));
```

Comment:
The dependency is still not injected. It should probably be `setSlug($slug, Slugify slugifier)` (but simulated with `func_get_args()`. I'm not sure of the implications.